flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/ClusterResourceStatisticsProvider.java

```diff
@@ -50,4 +52,7 @@
 
     /** Get profile of total pending resources. */
     ResourceProfile getPendingResource();
+
+    /** Get allocate slots information from the TaskManager with the given instance id. */
+    Collection<TaskManagerSlotInformation> getAllocatedSlotsOf(InstanceID instanceId);
```

Comment:
This is not **statistics**.

I'm not entirely sure whether we need to add this interface. Given that it is only used in `FineGrainedSlotManager#getAllocatedSlotsOf`, we may consider to move the implementation of `FineGrainedTaskManagerTracker#getAllocatedSlotsOf` to `FineGrainedSlotManager`.

I'm also fine with adding `getAllocatedSlotsOf` as an shortcut interface. In that case, we might consider to 1) move it to `TaskManagerResourceInfoProvider`, and 2) provide a default implementation basing on `getRegisteredTaskManager`.